pkg/rules/config.go

```diff
@@ -31,20 +31,17 @@ type Config struct {
 }
 
 func (cfg *Config) ContainsRules() bool {
-	if cfg.PrometheusConfig == nil {
-		return false
-	}
-	promCfg := cfg.PrometheusConfig
-	if len(promCfg.RuleFiles) == 0 {
-		log.Info("msg", "Rules files not found in the given Prometheus configuration file. Both rule-manager and alerting will not be initialized")
-		return false
+	if cfg.PrometheusConfig != nil && len(cfg.PrometheusConfig.RuleFiles) != 0 {
+		return true
 	}
+	return false
+}
 
-	if promCfg.AlertingConfig.AlertRelabelConfigs == nil && promCfg.AlertingConfig.AlertmanagerConfigs == nil {
-		// This is true when alerting config is not applied.
-		log.Info("msg", "Alerting configuration not present in the given Prometheus configuration file. Alerting will not be initialized")
+func (cfg *Config) ContainsAlertingConfig() bool {
+	if cfg.PrometheusConfig != nil && len(cfg.PrometheusConfig.AlertingConfig.AlertmanagerConfigs) != 0 {
```

Comment:
If `cfg.PrometheusConfig.AlertingConfig` is nil this will crash

Reply:
No it won't. In AND conditions, go does not move ahead with the next logic if the first one is false.